packages/ui/src/hooks/usePendingTx.tsx

```diff
@@ -1,39 +1,169 @@
 import { useCallback, useEffect, useMemo, useState } from 'react'
-import { useApi } from '../contexts/ApiContext'
-import { MultiProxy } from '../contexts/MultiProxyContext'
-import { MultisigStorageInfo } from '../types'
+import { useMultiProxy } from '../contexts/MultiProxyContext'
+import { HexString, MultisigStorageInfo } from '../types'
 import { useMultisigCallSubscription } from './useMultisigCallsSubscription'
 import { isEmptyArray } from '../utils'
 import { useAccountId } from './useAccountId'
+import { ApiPromise } from '@polkadot/api'
+import { useApi } from '../contexts/ApiContext'
+import { ISanitizedCall, getDisplayArgs, isProxyCall, parseGenericCall } from '../utils'
+import { GenericCall } from '@polkadot/types'
+import { AnyJson, AnyTuple } from '@polkadot/types/types'
+import dayjs from 'dayjs'
+import localizedFormat from 'dayjs/plugin/localizedFormat'
+dayjs.extend(localizedFormat)
 
 export interface PendingTx {
   from: string
   hash: string
   info: MultisigStorageInfo
 }
-export const usePendingTx = (multiProxy?: MultiProxy) => {
+
+export interface CallDataInfoFromChain {
+  callData?: HexString
+  hash?: string
+  name?: string
+  args?: AnyJson
+  info?: PendingTx['info']
+  from: string
+  timestamp?: Date
+  multiProxyAddress?: string
+}
+
+type AggGroupedByDate = { [index: string]: CallDataInfoFromChain[] }
+
+export const getMultisigInfo = (c: ISanitizedCall): Partial<CallDataInfoFromChain>[] => {
+  const result: Partial<CallDataInfoFromChain>[] = []
+
+  const getCallResult = (c: ISanitizedCall) => {
+    if (typeof c.method !== 'string' && c.method.pallet === 'multisig') {
```

Comment:
can we create a method isMultisigCall for example that the code will be self explanatory?

Reply:
This is only used here, and I'm not a fan of creating a specific function for a not reused one-liner.
This would open the door to then use such function everywhere, like at the next line to check whether it's a `multisig.asMulti`.. effectively opening a can of worms.